server/data/mappers/workAndInterestMapper.ts

```diff
@@ -0,0 +1,141 @@
+  const personalInterests: Array<PersonalInterest> = induction.personalSkillsAndInterests.interests || []
+  return {
+    skills: personalSkills.map(skill => skill.skillType).sort(enumComparator) || [],
+    otherSkill: personalSkills.find(skill => skill.skillType === 'OTHER')?.skillTypeOther,
+    personalInterests: personalInterests.map(interest => interest.interestType).sort(enumComparator),
+    otherPersonalInterest: personalInterests.find(interest => interest.interestType === 'OTHER')?.interestTypeOther,
+    updatedBy: induction.personalSkillsAndInterests.updatedBy,
+    updatedAt: moment(induction.personalSkillsAndInterests.updatedAt).toDate(),
+  }
+}
+
+const toWorkExperience = (induction: InductionResponse): WorkExperience => {
+  const previousJobs: Array<PreviousWorkExperience> = induction.previousWorkExperiences.experiences || []
+  return {
+    hasWorkedPreviously: induction.previousWorkExperiences.hasWorkedBefore,
+    jobs: previousJobs
+      ?.map(job => {
+        return {
+          type: job.experienceType,
+          other: job.experienceTypeOther,
+          role: job.role,
+          responsibilities: job.details,
+        }
+      })
+      .sort(jobComparator),
+    updatedBy: induction.previousWorkExperiences.updatedBy,
+    updatedAt: moment(induction.previousWorkExperiences.updatedAt).toDate(),
+  }
+}
+
+const toLongQuestionSetWorkInterests = (induction: InductionResponse): WorkInterests => {
+  const mostRecentlyUpdatedSection: 'MAIN_INDUCTION' | 'WORK_INTERESTS' = moment(induction.updatedAt).isSameOrAfter(
+    induction.futureWorkInterests.updatedAt,
+  )
+    ? 'MAIN_INDUCTION'
+    : 'WORK_INTERESTS'
+
+  const workInterests: Array<FutureWorkInterest> = induction.futureWorkInterests.interests || []
+  return {
+    hopingToWorkOnRelease: induction.workOnRelease.hopingToWork,
+    longQuestionSetAnswers: {
+      constraintsOnAbilityToWork: induction.workOnRelease.affectAbilityToWork.sort(enumComparator),
+      otherConstraintOnAbilityToWork: induction.workOnRelease.affectAbilityToWorkOther,
+      jobs: workInterests.map(workInterest => {
+        return {
+          jobType: workInterest.workType,
+          otherJobType: workInterest.workTypeOther,
+          specificJobRole: workInterest.role,
+        }
+      }),
+    },
+    shortQuestionSetAnswers: undefined,
+    updatedBy:
+      mostRecentlyUpdatedSection === 'MAIN_INDUCTION' ? induction.updatedBy : induction.futureWorkInterests.updatedBy,
+    updatedAt:
+      mostRecentlyUpdatedSection === 'MAIN_INDUCTION'
+        ? moment(induction.updatedAt).toDate()
+        : moment(induction.futureWorkInterests.updatedAt).toDate(),
+  }
+}
+
+const toShortQuestionSetWorkInterests = (induction: InductionResponse): WorkInterests => {
+  const mostRecentlyUpdatedSection: 'MAIN_INDUCTION' | 'WORK_INTERESTS' = moment(induction.updatedAt).isSameOrAfter(
+    induction.inPrisonInterests.updatedAt,
+  )
+    ? 'MAIN_INDUCTION'
+    : 'WORK_INTERESTS'
```

Comment:
Minor - is there any "common" code here that's worth putting into a separate method?

Reply:
Yeah, thats not a bad shout - I'll have a quick look at doing that 👍

Reply:
Ah, no, the duplicate sections are subtly different, so its probably not worth the effort. Perhaps I'll look at it again some other time